components/raftstore/src/store/worker/read.rs

```diff
@@ -782,6 +783,15 @@ where
             return Err(e);
         }
 
+        // Check witness
+        if find_peer_by_id(&delegate.region, delegate.peer_id)
```

Comment:
What if it's non-witness, and then change to witness? Is it possible the check pass but before taking a snapshot, data is destroyed?

Reply:
Seems possible, should recheck the `trackver` after the snapshot is taken. How do we avoid that for requesting a stale read while the peer is removed?

Reply:
It will double check safe_ts.

Reply:
Any update?

Reply:
fixed